src/Microsoft.Azure.WebJobs.Host/Blobs/Triggers/BlobTriggerAttributeBindingProvider.cs

```diff
@@ -18,14 +18,65 @@
 using Microsoft.Azure.WebJobs.Host.Triggers;
 using Microsoft.Extensions.Logging;
 using Microsoft.WindowsAzure.Storage.Blob;
+using Microsoft.Azure.WebJobs.Host.Config;
+using System.Threading;
 
 namespace Microsoft.Azure.WebJobs.Host.Blobs.Triggers
 {
+    internal class BlobTriggerExtension : IExtensionConfigProvider,
+        IAsyncConverter<IStorageBlob, Stream>
+    {
+        private IStorageAccountProvider _accountProvider;
+
+        public void Initialize(ExtensionConfigContext context)
+        {
+            _accountProvider = context.Config.GetService<IStorageAccountProvider>();
+
+            var rule = context.AddBindingRule<BlobTriggerAttribute>();
+            rule.BindToTrigger<IStorageBlob>();
```

Comment:
>BindToTrigger [](start = 17, length = 13)

This is new. This new rule allows triggers to leverage our converter pipeline (same as the regular BindToInput rules) and avoid having to implement their own value-providers.  @brettsam - CosmoDb may be able to benefit from this too.